### DISCUSSION TITLE
Are the Foundry apps anvil, forge and cast thread-safe at an app level?

### DISCUSSION THREAD

**Discussion by alandixon** [foundry-rs/foundry#9108] [Help]
Hi,
Are the Foundry apps anvil, forge and cast thread-safe?

Specifically, can I run multiple copies simultaneously on the same machine with no ill-effects?

The context is a CTF (Capture the flag) challenge where multiple users, probably numbering in the hundreds, will logon on to a Docker container and access Foundry via bash scripts.

So, each user will have their own foundry project but the Foundry apps could easily be creating projects, testing and running simultaneously.

Thanks

**Comment by zerosnacks** [comment]
Hi @alandixon 

One known area where this could be an issue are parallel solc installations

Foundry has not been specifically designed to be thread-safe. I am not aware if the Solidity compiler offers guarantees around thread safety.
